### PR TITLE
Specify that resolution of amount is msat

### LINKED ISSUE #692: Add pico amount example to Bolt11 test vectors
Several libraries (at least two libraries and one web service) have problems with the `p` numbers representing 10^(-12) BTC. So an example with this encoding should be included in the test vectors. You can use this one if you like:

```
lnbc9678785340p1pwmna7lpp5gc3xfm08u9qy06djf8dfflhugl6p7lgza6dsjxq454gxhj9t7a0sd8dgfkx7cmtwd68yetpd5s9xar0wfjn5gpc8qhrsdfq24f5ggrxdaezqsnvda3kkum5wfjkzmfqf3jkgem9wgsyuctwdus9xgrcyqcjcgpzgfskx6eqf9hzqnteypzxz7fzypfhg6trddjhygrcyqezcgpzfysywmm5ypxxjemgw3hxjmn8yptk7untd9hxwg3q2d6xjcmtv4ezq7pqxgsxzmnyyqcjqmt0wfjjq6t5v4khxxqyjw5qcqp2rzjq0gxwkzc8w6323m55m4jyxcjwmy7stt9hwkwe2qxmy8zpsgg7jcuwz87fcqqeuqqqyqqqqlgqqqqn3qq9qs4x9qlmd57lq7wwr23n3a6pkayy3jpfucyptlncs2maswe3dnnjy3ce2cgrvykmxlfpvn6ptqfqz4df5uaulvd4hjkckuqxrqqkz8jgphputwh
```

### PR DESCRIPTION
**Pull request by Sword-Smith**
When the `p` multiplier is used, the amount MUST be divisible
by 10 since the resolution used internally is millisatoshi.

This addresses but does not close-   #692.

### CODE REVIEW

**Comment by Sword-Smith** [comment]
I suggested the wording "MUST request `amount` divisible by ten if the `p` multiplier is used." but perhaps the wording "if the `p` multiplier is used `amount` MUST be divisible by ten."?

**Comment by cdecker** [comment]
ACK

File: 11-payment-encoding.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ A writer:
   - if a specific minimum `amount` is required for successful payment:
 	  - MUST include that `amount`.
 	- MUST encode `amount` as a positive decimal integer with no leading 0s.
+	- If the is `p` multiplier is used the `amount` the last decimal MUST be `0`.
```
Comment (t-bast) [inline 11-payment-encoding.md]: nit: 
```suggestion
	- If the `p` multiplier is used the `amount` the last decimal MUST be `0`.
```

**Comment by cdecker** [comment]
As discussed during the [spec meeting on Monday 2020/01/20](http://www.erisian.com.au/meetbot/lightning-dev/2020/lightning-dev.2020-01-20-19.06.html) I rephrased the requirement for trailing `0` decimal. I also added a short explanation to the rationale section, thus requiring new ACKs.

Ping @t-bast @Sword-Smith @Roasbeef @rustyrussell

**Comment by Sword-Smith** [comment]
I thought the wording was a bit off, made a fixup commit. Ping @cdecker @t-bast etc.

**Comment by cdecker** [comment]
Rebased and squashed

ACK https://github.com/lightningnetwork/lightning-rfc/commit/29f138699515d896b283da6fa53a5e16205eb2aa

File: 11-payment-encoding.md [hunk 2]
```diff
@@ -64,6 +64,7 @@ A writer:
   - if a specific minimum `amount` is required for successful payment:
 	  - MUST include that `amount`.
 	- MUST encode `amount` as a positive decimal integer with no leading 0s.
+	- If the `p` multiplier is used the `amount` the last decimal MUST be `0`.
```
Comment (t-bast) [inline 11-payment-encoding.md]: nit: @cdecker I think you missed a last comment that this sentence is weird.
It should be: "If the `p` multiplier is used the last decimal of the `amount` MUST be `0`."

Comment (cdecker) [reply 11-payment-encoding.md]: Oh damn, I thought there was a fixup from @Sword-Smith that fixed that. I'll open a new PR with the spelling mistake.

Comment (Sword-Smith) [reply 11-payment-encoding.md]: I think I did leave a fixup. But for some reason it didn't get included. I made a new PR
https://github.com/lightningnetwork/lightning-rfc/pull/733

**Pull request merged**